pkg/providers/amifamily/bootstrap/bottlerocket.go

```diff
@@ -76,6 +77,7 @@ func (b Bottlerocket) Script() (string, error) {
 	}
 
 	s.Settings.Kubernetes.NodeTaints = map[string][]string{}
+	b.Taints = append(b.Taints, v1.Taint{Key: "karpenter.sh/unregistered", Value: "true", Effect: v1.TaintEffectNoExecute})
```

Comment:
I'll update this to use the one in `kubernetes-sigs/karpenter` after it's merged.